### DISCUSSION TITLE
[rocblas] Install hangs

### DISCUSSION THREAD

**Discussion by Tacklebox** [rocm-arch/rocm-arch#703] [Support] [answered]
I don't think this is necessarily a bug, but I'm having trouble trying to install rocm from the aur.

It gets to the step
```
# Writing Kernels...
Generating kernels: Launching 12 threads...
Custom kernel filename /home/$username/.cache/yay/rocblas/src/build/library/src/build_tmp/TENSILE/assembly/DGEMM_Aldebaran_PKFixedAtomic512Latest.s
Custom kernel filename /home/$username//.cache/yay/rocblas/src/build/library/src/build_tmp/TENSILE/assembly/DGEMM_Aldebaran_PKFixedAtomic512_104.s
```
Which I think is during the installation of rocblas, at which point it sits there forever with a ton of ram and 0% cpu usage.

I can see a few `TensileCreateLibrary` processes sitting in btop.

I'm on kernel 5.16.11, installing with `yay` and if it makes a difference there is a 6800xt using the open source `amdgpu` driver in the system.

**Comment by bernharl** [comment]
Happened to me to, also on 5.16 with yay and an RX 6800xt. After ~40 ish minutes the install continued successfully

**Comment by throwm8** [comment, marked as answer]
Rocblas takes a lot of time to compile/install if you build it for all architectures like it does by default. You can edit the PKGBUILD and add -DAMDGPU_TARGETS=... where ... is your GPU arch. I have an RX6800 so I set it to gfx1030 and rocblas takes significantly less time to compile. You can find out your GPU rach with rocminfo.
